src/main/drivers/accgyro/accgyro_spi_bmi270.c

```diff
@@ -125,65 +125,62 @@ typedef enum {
 
 // BMI270 register reads are 16bits with the first byte a "dummy" value 0
 // that must be ignored. The result is in the second byte.
-static uint8_t bmi270RegisterRead(const busDevice_t *bus, bmi270Register_e registerId)
+static uint8_t bmi270RegisterRead(extDevice_t *dev, bmi270Register_e registerId)
 {
     uint8_t data[2] = { 0, 0 };
 
-    if (spiBusReadRegisterBuffer(bus, registerId, data, 2)) {
+    if (spiReadRegMskBufRB(dev, registerId, data, 2)) {
         return data[1];
     } else {
         return 0;
     }
 }
 
-static void bmi270RegisterWrite(const busDevice_t *bus, bmi270Register_e registerId, uint8_t value, unsigned delayMs)
+static void bmi270RegisterWrite(extDevice_t *dev, bmi270Register_e registerId, uint8_t value, unsigned delayMs)
 {
-    spiBusWriteRegister(bus, registerId, value);
+    spiWriteReg(dev, registerId, value);
     if (delayMs) {
         delay(delayMs);
     }
 }
 
 // Toggle the CS to switch the device into SPI mode.
 // Device switches initializes as I2C and switches to SPI on a low to high CS transition
-static void bmi270EnableSPI(const busDevice_t *bus)
+static void bmi270EnableSPI(extDevice_t *dev)
 {
-    IOLo(bus->busdev_u.spi.csnPin);
+    IOLo(dev->busType_u.spi.csnPin);
     delay(1);
-    IOHi(bus->busdev_u.spi.csnPin);
+    IOHi(dev->busType_u.spi.csnPin);
     delay(10);
 }
 
-uint8_t bmi270Detect(const busDevice_t *bus)
+uint8_t bmi270Detect(extDevice_t *dev)
 {
-    spiSetDivisor(bus->busdev_u.spi.instance, spiCalculateDivider(BMI270_MAX_SPI_CLK_HZ));
-    bmi270EnableSPI(bus);
+    spiSetClkDivisor(dev, spiCalculateDivider(BMI270_MAX_SPI_CLK_HZ));
+    bmi270EnableSPI(dev);
 
-    if (bmi270RegisterRead(bus, BMI270_REG_CHIP_ID) == BMI270_CHIP_ID) {
+    if (bmi270RegisterRead(dev, BMI270_REG_CHIP_ID) == BMI270_CHIP_ID) {
         return BMI_270_SPI;
     }
 
     return MPU_NONE;
 }
 
-static void bmi270UploadConfig(const busDevice_t *bus)
+static void bmi270UploadConfig(extDevice_t *dev)
```

Comment:
It looks like const correctness will have to be restored pretty much everywhere here.

Reply:
Done... I think...